src/bd/primitives/Primitive.java

```diff
@@ -0,0 +1,84 @@
+package bd.primitives;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import com.oracle.truffle.api.dsl.NodeFactory;
+
+
+/**
+ * Annotation to be applied on node classes that are meant to be used as primitive operations.
+ * Thus, the node implements a built-in function of the interpreter, and might need to be
+ * represented as a language-level method/function.
```

Comment:
Yes, the node is turned into a language-level method. There is a `#+` that is built using the node. I would expect that to be the same thing in TruffleMate. That's how it works in SOMns and TruffleSOM anyway. See https://github.com/SOM-st/TruffleSOM/blob/bd-primitivies/src/som/primitives/Primitives.java#L197 There an `SInvokable` is constructed from all nodes annotated with `@Primitive`.